stdlib/public/Concurrency/Task+TaskExecutor.swift

```diff
@@ -0,0 +1,212 @@
+      // we turn off the task executor preference for the scope of `operation`.
+      _getUndefinedTaskExecutor()
+    }
+
+  let record = _pushTaskExecutorPreference(taskExecutorBuiltin)
+  defer {
+    _popTaskExecutorPreference(record: record)
+  }
+
+  // No need to manually hop to the target executor, because as we execute
+  // the operation, its enqueue will respect the attached executor preference.
+
+  return try await operation()
+}
+
+/// Task with specified executor ----------------------------------------------
+
+@available(SwiftStdlib 9999, *)
+extension Task where Failure == Never {
+
+  /// Runs the given nonthrowing operation asynchronously
+  /// as part of a new top-level task on behalf of the current actor.
+  ///
+  /// This overload allows specifying a preferred ``TaskExecutor`` on which
+  /// the `operation`, as well as all child tasks created from this task will be
+  /// executing whenever possible. Refer to ``TaskExecutor`` for a detailed discussion
+  // of the effect of task executors on execution semantics of asynchronous code.
+  ///
+  /// Use this function when creating asynchronous work
+  /// that operates on behalf of the synchronous function that calls it.
+  /// Like `Task.detached(priority:operation:)`,
+  /// this function creates a separate, top-level task.
+  /// Unlike `Task.detached(priority:operation:)`,
+  /// the task created by `Task.init(priority:operation:)`
+  /// inherits the priority and actor context of the caller,
+  /// so the operation is treated more like an asynchronous extension
+  /// to the synchronous operation.
+  ///
+  /// You need to keep a reference to the task
+  /// if you want to cancel it by calling the `Task.cancel()` method.
+  /// Discarding your reference to a detached task
+  /// doesn't implicitly cancel that task,
+  /// it only makes it impossible for you to explicitly cancel the task.
+  ///
+  /// - Parameters:
+  ///   - executor: the preferred task executor for this task, and any child tasks created by it
+  ///   - priority: The priority of the task.
+  ///     Pass `nil` to use the priority from `Task.currentPriority`.
+  ///   - operation: The operation to perform.
+  @discardableResult
+  @_alwaysEmitIntoClient
+  public init(
+    on executor: any TaskExecutor,
+    priority: TaskPriority? = nil,
+    operation: __owned @Sendable @escaping () async -> Success
+  ) {
+    #if $BuiltinCreateAsyncTaskWithExecutor
+    // Set up the job flags for a new task.
+    let flags = taskCreateFlags(
+      priority: priority, isChildTask: false, copyTaskLocals: true,
+      inheritContext: true, enqueueJob: true,
+      addPendingGroupTaskUnconditionally: false,
+      isDiscardingTask: false)
+
+    // Create the asynchronous task.
+    let taskExecutorRef = executor.asUnownedTaskExecutor()
+    let (task, _) = Builtin.createAsyncTaskWithExecutor(
+      flags, taskExecutorRef.executor, operation)
```

Comment:
These `Task(on:)` are the reason for the `Builtin.createAsyncTaskWithExecutor` builtin. And the same is the case for the TaskGroup.addTask APIs